src/main/java/com/clover/habbittracker/domain/comment/service/CommentServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+package com.clover.habbittracker.domain.comment.service;
+
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.clover.habbittracker.domain.comment.dto.CommentRequest;
+import com.clover.habbittracker.domain.comment.dto.CommentResponse;
+import com.clover.habbittracker.domain.comment.entity.Comment;
+import com.clover.habbittracker.domain.comment.repository.CommentRepository;
+import com.clover.habbittracker.domain.member.entity.Member;
+import com.clover.habbittracker.domain.member.exception.MemberNotFoundException;
+import com.clover.habbittracker.domain.member.repository.MemberRepository;
+import com.clover.habbittracker.domain.post.entity.Post;
+import com.clover.habbittracker.domain.post.exception.PostNotFoundException;
+import com.clover.habbittracker.domain.post.repository.PostRepository;
+import com.clover.habbittracker.global.exception.PermissionDeniedException;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class CommentServiceImpl implements CommentService{
+
+	private final CommentRepository commentRepository;
+	private final MemberRepository memberRepository;
+	private final PostRepository postRepository;
+
+	@Override
+	@Transactional
+	public CommentResponse createComment(Long memberId, Long postId, CommentRequest request) {
+		Member member = getMemberBy(memberId);
+		Post post = getPostBy(postId);
+		Comment comment = Comment.builder()
+			.member(member)
+			.post(post)
+			.content(request.content())
+			.build();
+		Comment saveComment = commentRepository.save(comment);
+
+		return CommentResponse.from(saveComment);
+	}
+
+
+	@Override
+	@Transactional
+	public CommentResponse updateComment(Long memberId, Long commentId, Long postId, CommentRequest request) {
+		Comment comment = commentRepository.findByIdAndPostId(commentId, postId)
+			.orElseThrow(() -> new IllegalArgumentException("NotFoundComment"));
+		verifyPermissions(comment.getMember(), memberId);
+		comment.updateComment(request);
+		return CommentResponse.from(comment);
+	}
+
+	@Override
+	public List<CommentResponse> getReplyList(Long commentId, Long postId) {
+
+		List<Comment> childCommentList = commentRepository.findChildCommentById(commentId);
+		return CommentResponse.from(childCommentList);
+	}
+
+	@Override
+	@Transactional
+	public void createReply(Long memberId, Long commentId, Long postId, CommentRequest request) {
+		Member member = getMemberBy(memberId);
+		Post post = getPostBy(postId);
+		Comment comment = Comment.builder()
+			.member(member)
+			.post(post)
+			.parentId(commentId)
+			.content(request.content())
+			.build();
+
+		commentRepository.save(comment);
+	}
+
+	private void verifyPermissions(Member member, Long memberId) {
+		if (!member.getId().equals(memberId)) {
+			throw new PermissionDeniedException(memberId);
```

Comment:
```suggestion
		if (!Objects.equals(member.getId(), memberId)) {
			throw new PermissionDeniedException(memberId);
```

이런식으로 하면 NPE를 방지할 수 있습니다!

Reply:
memberId 가 Null 이 들어올거라는걸 생각도 못했네요!
감사합니다 해당 내용 수정했습니다!
[feat : CommentMapper 추가](https://github.com/Clover-Habiters/backend/pull/17/commits/c83221d1461d9197f52d866c4633fac9fc04cebd)

Reply:
여러 사람이 같이 코드를 작성하게 되면 언제 어떤 상황이 발생할 것이다 장담할 수 없게 됩니다! 그렇기 때문에 위험 요소는 최대한 배제하는 것이 좋습니다!